//
// Copyright 2021 Brad Post
//
syntax = "proto3";

package package_grpc.v1;

//
// Define the book categories
//
enum BookCategory {
    MYSTERY = 0;
    SCIENCE_FICTION = 1;
    SELF_HELP = 2;
}

//
// Define the API parameter
//
message RecommendationRequest {
    int32 user_id = 1;
    BookCategory category = 2;
    int32 max_results = 3;
}

//
// Define the Book Recommendation type
//
message BookRecommendation {
    int32 id = 1;
    string title = 2;
}

//
// Define the Recommendation Response
//
message RecommendationResponse {
    repeated BookRecommendation recommendations = 1;
}

//
// Define the API
//
service Recommendations {
    rpc Recommend (RecommendationRequest) returns (RecommendationResponse);
}
